src/rigid.rs

```diff
@@ -0,0 +1,261 @@
+use approxeq::ApproxEq;
+use num_traits::Float;
+use trig::Trig;
+use {TypedRotation3D, TypedTransform3D, TypedVector3D, UnknownUnit};
+
+/// A rigid transformation. All lengths are preserved under such a transformation.
+///
+///
+/// Internally, this is a rotation and a translation, with the rotation
+/// applied first (i.e. `Translation * Rotation`)
+///
+/// This can be more efficient to use over full matrices, especially if you
+/// have to deal with the decomposed quantities often.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[repr(C)]
+pub struct TypedRigidTransform3D<T, Src, Dst> {
+    pub rotation: TypedRotation3D<T, Src, Dst>,
+    pub translation: TypedVector3D<T, Dst>,
+}
+
+pub type RigidTransform3D<T> = TypedRigidTransform3D<T, UnknownUnit, UnknownUnit>;
+
+impl<T: Float + ApproxEq<T>, Src, Dst> TypedRigidTransform3D<T, Src, Dst> {
+    /// Construct a new rigid transformation, where the `rotation` applies first\
+    #[inline]
+    pub fn new(rotation: TypedRotation3D<T, Src, Dst>, translation: TypedVector3D<T, Dst>) -> Self {
```

Comment:
I kind of wonder if it might be easier if we had `from_translated_rotation` for this and `from_rotated_transform` for "reversed"? Up to you. I'm not sure whether the wordiness would help.